src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ReuploadFailedEnvelopeTask.java

```diff
@@ -55,29 +56,37 @@ public FailedDocUploadProcessor getProcessor() {
     @SchedulerLock(name = "re-upload-failures")
     @Scheduled(fixedDelayString = "${scheduling.task.reupload.delay}")
     public void processUploadFailures() throws InterruptedException {
-        ExecutorService executorService = Executors.newFixedThreadPool(
-            accessMapping.size(),
-            r -> new Thread(r, "BSP-REUPLOAD")
-        );
-        CompletionService<Void> completionService = new ExecutorCompletionService<>(executorService);
+        ExecutorService executorService = null;
 
-        accessMapping
-            .stream()
-            .map(Mapping::getJurisdiction)
-            .forEach(jurisdiction -> {
-                FailedDocUploadProcessor processor = getProcessor();
+        try {
+            executorService = Executors.newFixedThreadPool(
+                accessMapping.size(),
+                new ThreadFactoryBuilder().setNameFormat("BSP-REUPLOAD-%d").build()
```

Comment:
👍.

just a note for future us - we'll have to revisit google's code in near future as more and more libraries are beginning to upgrade their dependencies. i like to think in terms of owasp but that's slightly additional/other matter

Reply:
👍